lib/readline.js

```diff
@@ -954,102 +964,130 @@ exports.emitKeypressEvents = emitKeypressEvents;
 
 // Regexes used for ansi escape code splitting
 const metaKeyCodeReAnywhere = /(?:\x1b)([a-zA-Z0-9])/;
-const metaKeyCodeRe = new RegExp('^' + metaKeyCodeReAnywhere.source + '$');
 const functionKeyCodeReAnywhere = new RegExp('(?:\x1b+)(O|N|\\[|\\[\\[)(?:' + [
   '(\\d+)(?:;(\\d+))?([~^$])',
   '(?:M([@ #!a`])(.)(.))', // mouse
   '(?:1;)?(\\d+)?([a-zA-Z])'
 ].join('|') + ')');
```

Comment:
Yep, they are used in `stripVTControlCharacters` below.

I'm not sure whether it's a good idea to move them where they are [used](https://github.com/rlidwka/io.js/blob/fe3e0a0b2ca2970c2a0b73a0065e34c2732d59ef/lib/readline.js#L1428-L1431), or keep them near the comment where they are [explained](https://github.com/rlidwka/io.js/blob/fe3e0a0b2ca2970c2a0b73a0065e34c2732d59ef/lib/readline.js#L937-L963).